### PR TITLE
Normalize property reading from file and consul.

### PR DESCRIPTION
**Pull request by rabbah**
Reading properties from a file did not discriminate on the required+optional properties listed and read all properties in the file. This was different from reading from the environment or consul. Normalized so that file based reading of values is the same. Added additional tests and fixed a bug that caused values to be dropped when in the optional set.

### CODE REVIEW

**Comment by rabbah** [comment]
pg1/820 approved.

File: common/scala/src/main/scala/whisk/core/WhiskConfig.scala [hunk 1]
```diff
@@ -143,8 +144,8 @@ object WhiskConfig extends Logging {
     def readPropertiesFromConsul(properties: scala.collection.mutable.Map[String, String])(implicit system: ActorSystem) = {
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/WhiskConfig.scala]: it doesn't escape - updating in place is desirable?

**Comment by rabbah** [comment]
pg2/594 approved.

**Comment by sjfink** [comment]
rebased and merged 48a0362..3af6035

**Pull request closed without merging**